app/src/main/java/edu/unikom/herbamedjabar/repository/PlantRepository.kt

```
@@ -80,8 +118,16 @@ class PlantRepositoryImpl @Inject constructor(
             val directory = wrapper.getDir("images", android.content.Context.MODE_PRIVATE)
             val file = File(directory, "${UUID.randomUUID()}.jpg")
-            FileOutputStream(file).use { outputStream ->
-                bitmap.compress(Bitmap.CompressFormat.JPEG, 100, outputStream)
+            try {
+                FileOutputStream(file).use { outputStream ->
+                    check(bitmap.compress(Bitmap.CompressFormat.JPEG, COMPRESS_QUALITY, outputStream)) {
+                        application.getString(R.string.compression_failed)
+                    }
+                    outputStream.flush()
+                }
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)